### DISCUSSION TITLE
Are landmarks necessary to meet the requirements of 1.3.1 and 2.4.1?

### DISCUSSION THREAD

**Discussion by subbu-salesforce** [w3c/wcag#3570] [Q&A]
My webpage contains headings. Are landmarks necessary to meet the requirements of 1.3.1 and 2.4.1?

In my view, implementing landmarks is a best practice when each section is clearly delineated by headings. I would appreciate your thoughts on whether my understanding is correct.

**Comment by JAWS-test** [comment]
Landmarks are not necessary if all page areas have meaningful headings.

In 2.4.1, for example, landmarks and headings are given as 2 alternatives:
* ARIA11: Using ARIA landmarks to identify regions of a page (ARIA)
* H69: Providing heading elements at the beginning of each section of content (HTML)

**Comment by awkawk** [comment]
I'd just say "no, landmarks are not necessary", and neither are headings.  ARIA11 and H69 are just techniques and represent one way to meet the SC, but are not required either. Of course, headings are very useful and I do recommend them, but I can't say that they are required by WCAG 2.x.

**Reply by patrickhlauke** [reply]
I have a feeling we went around this particular bend a few times before. There *is* an argument that if a site has a clearly visually styled/delineated "header"/"footer"/"navigation"/"sidebar"/etc, a strict reading of 1.3.1 would suggest that this needs to be somehow conveyed in the underlying markup (either using a `<header>`/`<footer>`/etc element, or by conveying it with a `role="..."`) [edit: as @awkawk pointed out quite rightly: or conveyed somehow in text, whatever that means]. However, as this SC was devised before either of these were stable technologies, and since in many cases it can be clear just from context when part of a page is the header and when you're in the main part of it, etc ... a looser interpretation would say that these aren't required to pass the 1.3.1. don't think we ever got to a definitive answer on these...

**Comment by awkawk** [comment]
> strict reading of 1.3.1 would suggest that this needs to be somehow conveyed in the underlying markup

"or are available in text." is part of the text that we are reading strictly, yes?

**Reply by patrickhlauke** [reply]
well, what does that mean? an explicit mention in text ("This is the heading area", "This is the footer") ... or is it ok to have an implicit (based on context/content) understanding that you're now somewhere?

**Reply by awkawk** [reply]
Sure, it could mean that, and it could mean that it is apparent from the text more implicitly also.

Again, this isn't necessarily what I feel is best for good web content, but when I read statements like (sic) "landmarks aren't required as long as you have meaningful headings" I worry that people get the wrong message about what WCAG does and doesn't require.

**Reply by JAWS-test** [reply]
> "or are available in text." is part of the text that we are reading strictly, yes?

Text is provided as an exception if the information cannot be transmitted programmatically.

[Understanding](https://www.w3.org/WAI/WCAG22/Understanding/info-and-relationships.html):

> Some technologies do not provide a means to programmatically determine some types of information and relationships. In that case then there should be a text description of the information and relationships. 

> Situation B: The technology in use does NOT provide the semantic structure to make the information and relationships conveyed through presentation programmatically determinable: [G117: Using text to convey information that is conveyed by variations in presentation of text](https://www.w3.org/WAI/WCAG22/Techniques/general/G117) 

In Web, information on page areas can be transmitted programmatically. The means for this are landmarks or headings. If there are other means, these would also be permissible, but I don't know of any others.

**Reply by awkawk** [reply]
There is nothing in the SC (the normative part) that indicates that text is provided as an exception.

> Information, [structure](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-structure), and [relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-relationships) conveyed through [presentation](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-presentation) can be [programmatically determined](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html#dfn-programmatically-determined) or are available in text.
